inlong-sort/sort-single-tenant/src/test/java/org/apache/inlong/sort/singletenant/flink/parser/IcebergNodeSqlParserTest.java

```diff
@@ -0,0 +1,159 @@
+                .asList(new FieldRelationShip(new FieldInfo("id", new LongFormatInfo()),
+                                new FieldInfo("id", new LongFormatInfo())),
+                        new FieldRelationShip(new FieldInfo("name", new StringFormatInfo()),
+                                new FieldInfo("name", new StringFormatInfo())),
+                        new FieldRelationShip(new FieldInfo("age", new IntFormatInfo()),
+                                new FieldInfo("age", new IntFormatInfo())),
+                        new FieldRelationShip(new FieldInfo("ts", new TimestampFormatInfo()),
+                                new FieldInfo("ts", new TimestampFormatInfo()))
+                );
+
+        // set HIVE_CONF_DIR,or set uri and warehouse
+        Map<String, String> props = new HashMap<>();
+        props.put("catalog-type", "hive");
+        props.put("catalog-name", "hive_prod");
+        props.put("catalog-database", "default");
+        props.put("uri", "thrift://localhost:9083");
+        props.put("warehouse", "/hive/warehouse");
+        IcebergLoadNode node = new IcebergLoadNode("iceberg", "iceberg_output", fields, relations,
+                null, null, null, props, "inlong", "inlong_iceberg");
+        return node;
+    }
+
+    /**
+     * build node relation
+     *
+     * @param inputs extract node
+     * @param outputs load node
+     * @return node relation
+     */
+    private NodeRelationShip buildNodeRelation(List<Node> inputs, List<Node> outputs) {
+        List<String> inputIds = inputs.stream().map(Node::getId).collect(Collectors.toList());
+        List<String> outputIds = outputs.stream().map(Node::getId).collect(Collectors.toList());
+        return new NodeRelationShip(inputIds, outputIds);
+    }
+
+    @Test
+    public void testIceberg() throws Exception {
+        EnvironmentSettings settings = EnvironmentSettings
+                .newInstance()
+                .useBlinkPlanner()
+                .inStreamingMode()
+                .build();
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+        env.enableCheckpointing(10000);
+        StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env, settings);
+        Node inputNode = buildMySQLExtractNode("1");
+        Node outputNode = buildIcebergLoadNodeWithHiveCatalog();
+        StreamInfo streamInfo = new StreamInfo("1L", Arrays.asList(inputNode, outputNode),
+                Arrays.asList(buildNodeRelation(Collections.singletonList(inputNode),
+                        Collections.singletonList(outputNode))));
+        GroupInfo groupInfo = new GroupInfo("group_id", Collections.singletonList(streamInfo));
+        FlinkSqlParser parser = FlinkSqlParser.getInstance(tableEnv, groupInfo);
+        FlinkSqlParseResult result = parser.parse();
+        Assert.assertTrue(!result.getLoadSqls().isEmpty() && !result.getCreateTableSqls().isEmpty());

```

Comment:
Maybe it can be replaced by result.tryExecute()?

Reply:
no, insert iceberg table relay hive catlaog or hadoop catalog.
![image](https://user-images.githubusercontent.com/37214832/168581883-40c587a2-a4b2-459d-b632-a2a025bdd817.png)

Reply:
I am using testcontainers to support sort lightweight end2end test.After finish it, all sort source and sink can run in unit test.

Reply:
It is a very good idea.